File: dart/06-io-operations/io_output.dart
import 'dart:io';

void main() {
  // print() always appends a newline
  print('Standard output with print()');

  // stdout.write() does NOT append a newline
  stdout.write('No newline here... ');
  stdout.write('same line!\n');

  // writeln() writes its argument followed by a newline
  stdout.writeln('Written with writeln()');

  // stderr is a separate stream reserved for error/diagnostic output
  stderr.writeln('This goes to standard error');

  // writeAll() joins an iterable with a separator
  stdout.writeAll(['a', 'b', 'c'], '-');
  stdout.writeln();
}

File: nim/06-io-operations/io_output.nim
# echo joins all its arguments and adds a trailing newline
echo "echo adds a newline and can join args: ", 42, " ", true

# stdout.write does NOT add a newline
stdout.write("No newline here... ")
stdout.write("continued on same line\n")

# writeLine adds its own newline for you
stdout.writeLine("writeLine adds its own newline")

# write accepts multiple arguments in one call
stdout.write("a", "b", "c", "\n")
